crates/ruff_python_semantic/src/model.rs

```diff
@@ -546,6 +550,35 @@ impl<'a> SemanticModel<'a> {
         self.stmts.ancestor_ids(node_id).map(|id| self.stmts[id])
     }
 
+    /// Set the [`Globals`] for the current [`Scope`].
+    pub fn set_globals(&mut self, globals: Globals<'a>) {
```

Comment:
`set_*` feels bad, it'd be nice to instead provide these when we push the `Scope`, but that API only accepts a `ScopeKind` right now.